test/integrations/matching/DirectMatchTest.t.sol

```diff
@@ -0,0 +1,500 @@
+            ),
+            HEDGER
+        );
+
+        // Advance the time and accrue some interest. By fuzzing over the time
+        // to advance and the variable rate, we demonstrate that the orders that
+        // we create aren't "brittle." Once we find a set of orders that works
+        // to directly match two parties, the orders should still work several
+        // days later.
+        _timeElapsed = _timeElapsed.normalizeToRange(0, 3 days);
+        _variableRate = _variableRate.normalizeToRange(0, 0.1e18);
+        advanceTime(_timeElapsed, int256(_variableRate));
+
+        // Create the two orders and sign them.
+        IHyperdriveMatchingEngine.OrderIntent
+            memory longOrder = IHyperdriveMatchingEngine.OrderIntent({
+                trader: LP,
+                hyperdrive: hyperdrive,
+                amount: 2_440_000e6,
+                slippageGuard: 2_499_999e6,
+                minVaultSharePrice: hyperdrive
+                    .getCheckpoint(hyperdrive.latestCheckpoint())
+                    .vaultSharePrice,
+                options: IHyperdrive.Options({
+                    asBase: true,
+                    destination: LP,
+                    extraData: ""
+                }),
+                orderType: IHyperdriveMatchingEngine.OrderType.OpenLong,
+                signature: new bytes(0),
+                expiry: block.timestamp + 1 hours,
+                salt: bytes32(uint256(0xdeadbeef))
+            });
+        IHyperdriveMatchingEngine.OrderIntent
+            memory shortOrder = IHyperdriveMatchingEngine.OrderIntent({
+                trader: HEDGER,
+                hyperdrive: hyperdrive,
+                amount: 2_500_000e6,
+                slippageGuard: 65_000e6,
+                minVaultSharePrice: hyperdrive
+                    .getCheckpoint(hyperdrive.latestCheckpoint())
+                    .vaultSharePrice,
+                options: IHyperdrive.Options({
+                    asBase: true,
+                    destination: HEDGER,
+                    extraData: ""
+                }),
+                orderType: IHyperdriveMatchingEngine.OrderType.OpenShort,
+                signature: new bytes(0),
+                expiry: block.timestamp + 1 hours,
+                salt: bytes32(uint256(0xbeefbabe))
+            });
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(
+            LP_PK,
+            matchingEngine.hashOrderIntent(longOrder)
+        );
+        longOrder.signature = abi.encodePacked(r, s, v);
+        (v, r, s) = vm.sign(
+            HEDGER_PK,
+            matchingEngine.hashOrderIntent(shortOrder)
+        );
+        shortOrder.signature = abi.encodePacked(r, s, v);
+
+        // Update the short order's extra data after signing. This simulates
+        // what the DFB UI will do to ensure an up-to-date quote. Since the
+        // extra data isn't baked into the hash, this will not cause issues with
+        // signature verification.
```

Comment:
NIT: This tidbit about `extraData`'s relevance to the DFB UI might be a nice addition to the interface... seeing that it wasn't included in the hash made me think it wasn't relevant or was being ignored.